river/Root.zig

```diff
@@ -128,7 +128,15 @@ pub fn init(self: *Self) !void {
     hidden_tree.node.setEnabled(false);
 
     const outputs = try interactive_content.createSceneTree();
-    const xwayland_override_redirect = if (build_options.xwayland) try interactive_content.createSceneTree();
+    const xwayland_override_redirect = blk: {
+        if (build_options.xwayland) {
+            if (server.xwayland != null) {
+                break :blk try interactive_content.createSceneTree();
+            } else {
+                break :blk null;
+            }
+        }
```

Comment:
```suggestion
        if (build_options.xwayland and server.xwayland != null) {
            break :blk try interactive_content.createSceneTree();
        } else {
            break :blk null;
        }
```

Reply:
If I do this, if `build_options.xwayland` is false it means that we have a type null instead of void

Reply:
Should be good to land then :)